roachtest: version/mixedWith* log spam

Last night's [roachtest nightly run](https://teamcity.cockroachdb.com/viewLog.html?buildId=944756&tab=artifacts&buildTypeId=Cockroach_Nightlies_WorkloadNightly&showAll=true) for the `version/mixedWith` tests produced enormous `test.log` files:

```
-rw-r--r-- 1 root root 7.7G Oct  5 21:09 './artifacts/20181005-944756/version/mixedWith=v2.0.5/nodes=3/test.log'
-rw-r--r-- 1 root root 4.3G Oct  5 21:23 './artifacts/20181005-944756/version/mixedWith=v2.0.5/nodes=5/test.log'
```

Examining the files shows that the log message `workload 1: E181005 20:46:28.835841 1 workload/cli/run.go:396  database/sql: connection is already closed` is repeated over and over. Not sure why this happened yet. The large `test.log` files caused the agent to run out of disk space causing failure in subsequent roachtests. See https://github.com/cockroachdb/cockroach/issues/31027.

From another nightly run, it seems that something is fouled up with the teardown in this test. The error is different, and not repeated as much: `E181004 17:20:57.742138 1 workload/cli/run.go:395  error in payment: pq: server is not accepting clients`.

--- a/pkg/cmd/roachtest/version.go
+++ b/pkg/cmd/roachtest/version.go
@@ -70,7 +70,9 @@ func registerVersion(r *registry) {
 			i := i     // ditto
 			m.Go(func(ctx context.Context) error {
 				cmd = fmt.Sprintf(cmd, nodes)
-				childL, err := c.l.ChildLogger("workload " + strconv.Itoa(i))
+				// Direct stderr only to disk. We expect errors from the workload as
+				// nodes are stopped and started.
+				childL, err := c.l.ChildLogger("workload"+strconv.Itoa(i), quietStderr)
 				if err != nil {
 					return err
 				}

--- a/pkg/workload/cli/run.go
+++ b/pkg/workload/cli/run.go
@@ -388,12 +388,15 @@ func runRun(gen workload.Generator, urls []string, dbName string) error {
 		jsonEnc = json.NewEncoder(jsonF)
 	}
 
+	everySecond := log.Every(time.Second)
 	for i := 0; ; {
 		select {
 		case err := <-errCh:
 			numErr++
 			if *tolerateErrors {
-				log.Error(ctx, err)
+				if everySecond.ShouldLog() {
+					log.Error(ctx, err)
+				}
 				continue
 			}
 			return err

